### DISCUSSION TITLE
Syntax highlighting for inline code/code block

### DISCUSSION THREAD

**Discussion by yoonwaiyan** [athensresearch/athens#758] [Feature Requests]
This has become a pretty standard feature for any editor that supports inline code and code blocks. Would be delighted to see this being implemented in Athens. There are some native JS libraries to support syntax highlighting, which could help reduce the development effort needed for this. Besides, I think the parsing for code blocks needs to be improved, as I see from the screen that when I input a language in an attempt to try out whether there's any syntax highlighting, the code block is displaying the language in the code block for reading mode:

Edit mode:
<img width="678" alt="Screenshot 2021-03-06 at 4 55 44 PM" src="https://user-images.githubusercontent.com/3081772/110201200-d5173d00-7e9c-11eb-9ffa-57b117946043.png">

Reading mode:
<img width="525" alt="Screenshot 2021-03-06 at 4 55 38 PM" src="https://user-images.githubusercontent.com/3081772/110201210-de080e80-7e9c-11eb-82e6-fef10c26666f.png">

Edit: I also realized multi-line blocks are displayed as one-liner instead, and pasting from external sources created multiple editor blocks instead of being pasted properly inside the code block.

**Comment by dscamp** [comment]
I'd also like to see the block highlighted while in reading mode.  Good examples are at https://markdownlivepreview.com/
![image](https://user-images.githubusercontent.com/3029759/111405030-89d80800-86ae-11eb-9545-317072f7015c.png)
